app/test/processing/app/SerialTest.java

```diff
@@ -51,8 +51,16 @@ protected void message(char[] chars, int length) {
   public void testSerialUTF8Decoder() throws Exception {
     NullSerial s = new NullSerial();
     // https://github.com/arduino/Arduino/issues/9808
-    String testdata = "012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789°0123456789";
-    s.processSerialEvent(testdata.getBytes());
-    assertEquals(s.output, testdata);
+    byte testData[] = { 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 49, 50, 51,
+        52, 53, 54, 55, 56, 57, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 49,
+        50, 51, 52, 53, 54, 55, 56, 57, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57,
+        48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 49, 50, 51, 52, 53, 54, 55,
+        56, 57, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 49, 50, 51, 52, 53,
+        54, 55, 56, 57, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 48, 49, 50, 51,
+        52, 53, 54, 55, 56, 57, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, //
+        -62, -80, // UTF8 char
+        48, 49, 50, 51, 52, 53, 54, 55, 56, 57 };
+    s.processSerialEvent(testData);
+    assertEquals(new String(testData, "UTF-8"), s.output);
```

Comment:
Hm, this is a lot less readable. Can you not fix this by passing a charset to `getBytes()` explicitly? https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#getBytes(java.nio.charset.Charset)

Reply:
I'll fix this in an incoming PR.